docs/src/metrics.rst

```diff
@@ -4,8 +4,64 @@
 Metrics operations
 ======================
 
-libuv provides a metrics API to track the amount of time the event loop has
-spent idle in the kernel's event provider.
+libuv provides a metrics API to track various internal operations of the event
+loop.
+
+
+Data types
+----------
+
+.. c:type:: uv_metrics_t
+
+    Metrics handle type.
+
+.. c:type:: uv_metrics_list_t
+
+    The struct of metrics that are passed to :c:type:`uv_metrics_cb`. These
+    metrics start being tracked internally when the :c:type:`uv_loop_t` is
+    initialized. So any call to the :c:type:`uv_metrics_cb` will have the most
+    recent metrics. This is useful when metrics only need to be sampled at
+    intervals.
+
+    ::
+
+        typedef struct {
+            uint64_t loop_count;
+            uint64_t events;
+            uint64_t events_waiting;
+        } uv_metrics_list_t;
```

Comment:
A question: in case we'd want to expand the list of metrics, would it break ABI compatibility?

Reply:
I guess it would if the user maintained a stack allocated instance. You think I should add reserved fields in case we want to extend these in the future?

Reply:
Reserved fields.